### PR TITLE
Support container_push on Windows.

### PR DESCRIPTION
**Pull request by faube**
- Use a batch file instead of a shell script.
- Fix paths by using backslashes.

### CODE REVIEW

File: container/push.bzl [hunk 1]
```diff
@@ -205,3 +215,26 @@ container_push = rule(
...
+            "//conditions:default": "",
+        }),
+        tag_tpl = select({
+            "@bazel_tools//src/conditions:host_windows": Label("//container:push-tag.bat.tpl"),
+            "//conditions:default": Label("//container:push-tag.sh.tpl"),
+        }),
+        windows_paths = select({
+            "@bazel_tools//src/conditions:host_windows": True,
+            "//conditions:default": False,
+        }),
```
Comment (smukherj1) [inline container/push.bzl:238]: image

**Review by smukherj1 — APPROVED** [review]

File: container/push.bzl (hunk 1 above)
Comment (smukherj1) [reply container/push.bzl:238]: More like I have no idea how this bit works

**Comment by faube** [comment]
Looking.

**Comment by faube** [comment]
I think this is the error. Could it be transient? Is there a way to retry
the build kite? In my VM I can run this no problem: bazel build
@alpine_linux_armv6_fixed_id//image


[1A [K(14:15:04) [31m [1mERROR: [0mAn error occurred during the fetch of
repository 'alpine_linux_armv6_fixed_id':

   Traceback (most recent call last):

File "/workdir/container/pull.bzl", line 171, column 13, in _impl

fail("Pull command failed: %s (%s)" % (result.stderr, " ".join([str(a) for
a in args])))

Error in fail: Pull command failed: 2020/10/01 14:15:03 Running the Image
Puller to pull images from a Docker Registry...

2020/10/01 14:15:04 Image pull was unsuccessful: reading image "
index.docker.io/library/alpine@sha256:f29c3d10359dd0e6d0c11e4f715735b678c0ab03a7ac4565b4b6c08980f6213b":
GET
https://auth.docker.io/token?scope=repository%3Alibrary%2Falpine%3Apull&service=registry.docker.io:
unsupported status code 500

 (/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/ec321eb2cc2d0f8f91b676b6d4c66c29/external/go_puller_linux/file/downloaded
-directory
/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/ec321eb2cc2d0f8f91b676b6d4c66c29/external/alpine_linux_armv6_fixed_id/image
-os linux -os-version  -os-features  -architecture arm -variant v6
-features  -name
index.docker.io/library/alpine@sha256:f29c3d10359dd0e6d0c11e4f715735b678c0ab03a7ac4565b4b6c08980f6213b
)

**Comment by smukherj1** [comment]
retrying

**Pull request merged**